Epam/Java Basics_UpSkill Lab #1/Strings_and_basics_of_text_processing/Working_with_a_string_as_an_array_of_characters/src/com/androsor/string_char/HomeTas2.java

```diff
@@ -0,0 +1,24 @@
+package com.androsor.string_char;
+
+/**
+ * Замените в строке все вхождения 'word' на 'letter'.
+ */
+
+public class HomeTas2 {
+
+    public static void main(String[] args) {
+
+        String strInitial;
+        String strWordOne;
+        String strWordTwo;
+
+        strInitial = HomeTask1.enterFromConsole("Ведите исходную строку");
+        strWordOne = HomeTask1.enterFromConsole("Ведите заменяемое слово");
+        strWordOne = strWordOne.trim();
+        strWordTwo = HomeTask1.enterFromConsole("Ведите заменяющее слово");
+        strWordTwo = strWordTwo.trim();
+
+        System.out.printf("Замененная строка \n %s", strInitial.replace(strWordOne, strWordTwo));
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.